app/models/mixins/aggregation_mixin.rb

```diff
@@ -95,7 +95,8 @@ def aggregate_hardware(from, field, targets = nil)
     select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
     targets ||= send("all_#{from}_ids")
     targets   = targets.collect(&:id) unless targets.first.kind_of?(Integer)
-    hdws      = Hardware.where("#{from}_id" => targets).select(select)
+    column_name = Hardware.column_names.detect { |col| col == "#{from}_id" }
```

Comment:
I think add a comment here, because as a casual coder I would see this and just straight up remove it.

Also, is there a way to use `Hardware.columns_hash` so that you can do an O(1) lookup as opposed to O(N).  Maybe `Hardware.columns_hash["#{from}_id"].name`?